crystal_diffusion/train_mtp.py

```diff
@@ -0,0 +1,202 @@
+            structures.append(pm_structure)
+            force_idx = [d['keywords'].index(x) for x in ['fx', 'fy', 'fz']]
+            structure_forces = [[x[i] for i in force_idx] for x in d['data']]
+            forces.append(structure_forces)
+    return structures, forces
+
+
+def extract_energy_from_thermo_log(filename: str) -> List[float]:
+    """Read energies from LAMMPS thermodynamic output file.
+
+    Args:
+        filename: path to LAMMPS thermodynamic output file in yaml format.
+
+    Returns:
+        list of energies (1 value per configuration)
+    """
+    with open(filename, 'r') as f:
+        log_yaml = yaml.safe_load(f)
+        kin_idx = log_yaml['keywords'].index('KinEng')
+        pot_idx = log_yaml['keywords'].index('PotEng')
+        energies = [x[kin_idx] + x[pot_idx] for x in log_yaml['data']]
+    return energies
+
+
+def prepare_mtp_inputs_from_lammps(output_yaml: List[str],
+                                   thermo_yaml: List[str],
+                                   atom_dict: Dict[int, Any]
+                                   ) -> Dict[str, Any]:
+    """Convert a list of LAMMPS output files and thermodynamic output files to MTP input format.
+
+    Args:
+        output_yaml: list of LAMMPS output files as yaml.
+        thermo_yaml: list of LAMMPS thermodynamic output files as yaml.
+        atom_dict: mapping of LAMMPS indices to atom type.
+
+    Returns:
+        dict with structure, energies and forces usable by MTP.
+    """
+    mtp_inputs = {
+        'structure': [],
+        'energy': [],
+        'forces': []
+    }
+    for filename in output_yaml:
+        structures, forces = extract_structure_and_forces_from_file(filename, atom_dict)
+        mtp_inputs['structure'] += structures
+        mtp_inputs['forces'] += forces
+    for filename in thermo_yaml:
+        mtp_inputs['energy'] += extract_energy_from_thermo_log(filename)
+    return mtp_inputs
+
+
+def train_mtp(train_inputs: Dict[str, Any], mlip_cmd_path: str, save_dir: str) -> MTPWithMLIP3:
+    """Create and evaluate an MTP potential.
+
+    Args:
+        train_inputs: inputs for training. Should contain structure, energies and forces
+        mlip_cmd_path: path to MLIP-3 folder
+        save_dir: path to directory where to save the fitted model
+
+    Returns:
+       dataframe with original and predicted energies and forces.
+    """
+    # TODO more kwargs for MTP training. See maml and mlip-3 documentation.
+    # create MTP
+    mtp = MTPWithMLIP3(mlip_path=mlip_cmd_path)
+    # train
+    mtp.train(
+        train_structures=train_inputs["structure"],
+        train_energies=train_inputs["energy"],
+        train_forces=train_inputs["forces"],
+        train_stresses=None,
+        max_dist=5,
+        stress_weight=0,
+        fitted_mtp_savedir=save_dir,
+    )
+
+    return mtp
+
+
+def evaluate_mtp(eval_inputs: Dict[str, Any], mtp: MTPWithMLIP3) -> Tuple[pd.DataFrame, pd.DataFrame]:
```

Comment:
I think `eval_inputs` should be a `namedtuple`.

Reply:
changed